File: Lecture09/AsynchronousWork/app/src/main/java/com/codingblocks/asynchronouswork/MainActivity.java
package com.codingblocks.asynchronouswork;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnStart = findViewById(R.id.btnStart);
        final TextView tvDone = findViewById(R.id.tvDone);

        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Thread t = new Thread(new Runnable() {
                    @Override
                    public void run() {
                        for (long i = 0; i < 100000000L; i++) {
                            //Do nothing
                        }

                        MainActivity.this.runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                tvDone.setText("Counting Done!");
                            }
                        });
//                        tvDone.setText("Counting Done!");
//                        Toast.makeText(MainActivity.this, "Counting Done", Toast.LENGTH_SHORT).show();
                    }
                });

                t.start();

            }
        });
    }
}

File: Lecture15/AlertDialogs/app/src/main/java/com/codingblocks/alertdialogs/MainActivity.kt
package com.codingblocks.alertdialogs

import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.widget.EditText
import android.widget.Toast
import android.widget.Toast.LENGTH_SHORT
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.alert_custom.*
import kotlinx.android.synthetic.main.alert_custom.view.*

class MainActivity : AppCompatActivity() {

    private val itemsArray = arrayOf("Option 1", "Option 2", "Option 3")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val simpleAlert: AlertDialog = AlertDialog.Builder(this)
            .setMessage("I'm a simple Alert Dialog")
            .setTitle("Simple Alert Dialog")
//            .setItems(arrayOf("Option 1", "Option 2", "Option 3")) { dialog, which ->
//
//            }
            .setCancelable(false)
            .setNegativeButton("No") { dialog, which ->
                Toast.makeText(this, "Negative button was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }
            .setPositiveButton("Yes") { dialog, which ->
                Toast.makeText(this, "Positive button was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }
            .create()


        val listAlert: AlertDialog = AlertDialog.Builder(this)
//            .setMessage("I'm a list Alert Dialog")
            .setTitle("List Alert Dialog")
            .setItems(itemsArray) { dialog, which ->
                Toast.makeText(this, "The item ${itemsArray[which]} was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }

            .setCancelable(false)
            .setNegativeButton("No") { dialog, which ->
                Toast.makeText(this, "Negative button was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }
            .setPositiveButton("Yes") { dialog, which ->
                Toast.makeText(this, "Positive button was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }
            .create()


        val inflatedView = LayoutInflater.from(this).inflate(R.layout.alert_custom, null, false)

        val customAlert: AlertDialog = AlertDialog.Builder(this)
            .setView(inflatedView)
            .setTitle("Create a new Note")
            .setNegativeButton("No") { dialog, which ->
                Toast.makeText(this, "Negative button was clicked", LENGTH_SHORT).show()
                dialog.dismiss()
            }
            .setPositiveButton("Yes") { dialog, which ->
                Toast.makeText(
                    this,
                    "You entered ${inflatedView.etTitle.text} and ${inflatedView.etDesc.text}",
                    LENGTH_SHORT
                ).show()
                //Save this to the DB
                dialog.dismiss()
            }
            .create()

        btnSimpleDialog.setOnClickListener {
            //Show a simple alertDialog here
            simpleAlert.show()
        }

        btnListDialog.setOnClickListener {
            listAlert.show()
        }

        btnCustomDialog.setOnClickListener {
            //Show a custom alertDialog here
            customAlert.show()
        }

    }
}
